src/catalog/models/base_model.py

```diff
@@ -74,9 +74,12 @@ def update(self, **kwargs):
 
     def delete(self):
         """Delete and commit"""
-
-        db_session.delete(self)
-        self._commit()
+        try:
```

Comment:
Add test case for this function. I think we don't have any unit testing for this file. Can you please create on and add test cases?

Reply:
Yes we will need to create testcases for Base Model. Will add that